Projects/Delivery/ApiHandlers/src/com/ilmlf/delivery/api/handlers/PopulateFarmDb.java

```diff
@@ -0,0 +1,92 @@
+package com.ilmlf.delivery.api.handlers;
+
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.CloudFormationCustomResourceEvent;
+import com.ilmlf.delivery.api.handlers.util.DbUtil;
+import com.ilmlf.delivery.api.handlers.util.SecretsUtil;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLSyntaxErrorException;
+import java.sql.Statement;
+import org.json.JSONObject;
+
+/**
+ * Seeds the database.
+ */
+public class PopulateFarmDb implements RequestHandler<CloudFormationCustomResourceEvent, String> {
+  static String DB_ENDPOINT = System.getenv("DB_ENDPOINT");
+  static String DB_REGION = System.getenv("DB_REGION");
+  static JSONObject DB_ADMIN_SECRET = SecretsUtil.getSecret(DB_REGION, System.getenv("DB_ADMIN_SECRET"));
+  static String DB_ADMIN_USER = (String) DB_ADMIN_SECRET.get("username");
+  static String DB_ADMIN_PWD = (String) DB_ADMIN_SECRET.get("password");
+  static JSONObject DB_USER_SECRET = SecretsUtil.getSecret(DB_REGION, System.getenv("DB_USER_SECRET"));
+  static Connection con;
+
+  static {
+    try {
+      con = DbUtil.createConnectionViaUserPwd(DB_ADMIN_USER, DB_ADMIN_PWD, DB_ENDPOINT);
+    } catch (Exception e) {
+      System.out.println("INIT connection FAILED");
+      System.out.println(e.toString());
+    }
+  }
+
+  @Override
+  public String handleRequest(CloudFormationCustomResourceEvent event, Context context) {
+    try {
+      String script = "com/ilmlf/delivery/api/handlers/db/dbinit.sql";
+      FileReader fr = new FileReader(script);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fr*. The resource is referenced at lines 43, 48. The resource is closed at line 68. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *fr* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).